src/test/java/duke/task/DukeListTest.java

```diff
@@ -0,0 +1,46 @@
+package duke.task;
+
+import duke.task.DukeList;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import static org.junit.jupiter.api.Assertions.*;
+
+public class DukeListTest {
+    private DukeList dukeList;
+
+    @BeforeEach
+    public void setup() {
+        dukeList = new DukeList(new ArrayList<>());
+    }
+
+    @Test
+    public void testAddTodo() {
```

Comment:
Not sure I like how the test cases are named. It might be better to follow the [textbook's naming conventions](https://nus-cs2103-ay2324s1.github.io/website/schedule/week3/topics.html#w3-7-unit-testing).
```suggestion
    public void addTodo_validInput_successfullyAddedToList() {
```
I do like how comprehensive the tests are though, most of the DukeList functions are tested here.